core/src/main/java/com/datastrato/gravitino/storage/relational/session/SqlSessions.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.storage.relational.session;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.ibatis.session.SqlSession;
+import org.apache.ibatis.session.TransactionIsolationLevel;
+
+/**
+ * SqlSessions is a utility class to maintain the MyBatis's {@link SqlSession} object. It is a
+ * thread local class and should be used to get the {@link SqlSession} object. It also provides the
+ * methods to commit, rollback and close the {@link SqlSession} object.
+ */
+public final class SqlSessions {
+  private static final ThreadLocal<SqlSession> sessions = new ThreadLocal<>();
+
+  private SqlSessions() {}
+
+  @VisibleForTesting
+  static ThreadLocal<SqlSession> getSessions() {
+    return sessions;
+  }
+
+  /**
+   * Get the SqlSession object. If the SqlSession object is not present in the thread local, then
+   * create a new SqlSession object and set it in the thread local.
+   *
+   * @return SqlSession object from the thread local storage.
+   */
+  public static SqlSession getSqlSession() {
+    SqlSession sqlSession = sessions.get();
+    if (sqlSession == null) {
+      sqlSession =
+          SqlSessionFactoryHelper.getInstance()
+              .getSqlSessionFactory()
+              .openSession(TransactionIsolationLevel.READ_COMMITTED);
+      sessions.set(sqlSession);
+      return sqlSession;
+    }
+    return sqlSession;
+  }
+
+  /**
+   * Commit the SqlSession object and close it. It also removes the SqlSession object from the
+   * thread local storage.
+   */
+  public static void commitAndCloseSqlSession() {
+    SqlSession sqlSession = sessions.get();
+    if (sqlSession != null) {
+      sqlSession.commit();
+      sqlSession.close();
+      sessions.remove();
+    }
+  }
+
+  /**
+   * Rollback the SqlSession object and close it. It also removes the SqlSession object from the
+   * thread local storage.
+   */
+  public static void rollbackAndCloseSqlSession() {
+    SqlSession sqlSession = sessions.get();
+    if (sqlSession != null) {
+      sqlSession.rollback();
+      sqlSession.close();
```

Comment:
It worries me that if "commit" and " roll back" both fail in some instances, such as when the MySQL server is suddenly inaccessible, the "sqlSession" will not be closed.

Reply:
The SqlSession extends the `Closeable` interface, if both `commit` and `rollback` throw exceptions, we use the `try-with-resource` exception handling mechanism in the `SessionUtils` class to ensure that the `SqlSession` can be closed eventually